exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -36,13 +36,25 @@ type kafkaTracesProducer struct {
 	logger    *zap.Logger
 }
 
+type kafkaErrors struct {
+	count int
+	err   string
+}
+
+func (ke kafkaErrors) Error() string {
+	return fmt.Sprintf("Failed to deliver %d messages due to %s", ke.count, ke.err)
+}
+
 func (e *kafkaTracesProducer) tracesPusher(_ context.Context, td pdata.Traces) error {
 	messages, err := e.marshaler.Marshal(td, e.topic)
 	if err != nil {
 		return consumererror.Permanent(err)
 	}
 	err = e.producer.SendMessages(messages)
 	if err != nil {
+		if value, ok := err.(sarama.ProducerErrors); ok {
+			return kafkaErrors{len(value), value[0].Err.Error()}
```

Comment:
Does accessing the first element may cause a crash? We need to check the length correct?

Reply:
The error will be returned by prouder  when it's length > 0. 
```
	if len(errors) > 0 {
		return errors
	}
```
But just to be sure is ok, I'd better check it. I have add the check, plz review it.